apps/frontend/src/pages/analytics/analytics.tsx

```diff
@@ -1,28 +1,73 @@
-import { DateInput, PageLayout } from "~/libs/components/components.js";
+import { DateInput, Loader, PageLayout } from "~/libs/components/components.js";
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import { DataStatus } from "~/libs/enums/enums.js";
 import { subtractDays } from "~/libs/helpers/helpers.js";
-import { useAppForm, useCallback } from "~/libs/hooks/hooks.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useState,
+} from "~/libs/hooks/hooks.js";
+import { actions as activityLogActions } from "~/modules/activity/activity.js";
 
+import { AnalyticsTable } from "./libs/components/components.js";
 import { ANALYTICS_DATE_MAX_RANGE } from "./libs/constants/constants.js";
 import styles from "./styles.module.css";
 
 const Analytics = (): JSX.Element => {
+	const dispatch = useAppDispatch();
 	const todayDate = new Date();
 
-	const { control, handleSubmit } = useAppForm({
+	const initialDateRange: [Date, Date] = [
+		subtractDays(todayDate, ANALYTICS_DATE_MAX_RANGE),
+		todayDate,
+	];
+
+	const { activityLogs, dataStatus } = useAppSelector(
+		({ activityLogs }) => activityLogs,
+	);
+
+	const [dateRange, setDateRange] = useState<[Date, Date]>(initialDateRange);
+
+	const { control, handleSubmit, isDirty, watch } = useAppForm({
 		defaultValues: {
-			dateRange: [subtractDays(todayDate, ANALYTICS_DATE_MAX_RANGE), todayDate],
+			dateRange: initialDateRange,
 		},
 	});
 
+	const handleLoadActivityLogs = useCallback(() => {
+		void dispatch(activityLogActions.loadAll());
+	}, [dispatch]);
```

Comment:
I think we can call `dispatch` in `useEffect` directly, can't we? We won't reuse this function anywhere else and it is just an extra code.